### ISSUE TITLE
9.1.5: GetOrCreateTableEntry misbehaves if defaultValue is false

### ISSUE THREAD

**Issue by Meorawr** [Stanzilla/WoWUIBugs#176]
Patch 9.1.5.40290 adds a new table utility function defined as follows:

```lua
function GetOrCreateTableEntry(table, key, defaultValue)
	local currentValue = table[key];
	if currentValue == nil then
		currentValue = defaultValue or {};
		table[key] = currentValue;
	end

	return currentValue;
end
```

If the value passed as `defaultValue` is boolean `false` and no existing value is present in the table for the given key, this will incorrectly default the value to a newly created table. This can be tested with the following, which will raise a script error:

```
/run GetOrCreateTableEntry(_G, "TestEntry", false); assert(TestEntry == false, "'TestEntry' was not defaulted to false");
```

**Comment by Meorawr** [comment]
Fixed internally, pending a future build.

**Comment by Meorawr** [comment]
Fixed in 9.1.5.40383.
